pkg/pool-weighted/test/WeightedPoolProtocolFees.behavior.ts

```diff
@@ -98,6 +103,111 @@ export function itPaysProtocolFeesFromInvariantGrowth(): void {
       });
     });
 
+    describe('ath rate product', () => {
+      context('when the pool is exempt from yield fees', () => {
+        let yieldFeeExemptPool: WeightedPool;
+
+        sharedBeforeEach(async () => {
+          yieldFeeExemptPool = await WeightedPool.create({
+            poolType: WeightedPoolType.WEIGHTED_POOL,
+            tokens,
+            weights: WEIGHTS.slice(0, numTokens),
+            swapFeePercentage: POOL_SWAP_FEE_PERCENTAGE,
+          });
+        });
+
+        it('is not set on initialization', async () => {
+          await yieldFeeExemptPool.init({ initialBalances });
+
+          expect(await yieldFeeExemptPool.instance.getATHRateProduct()).to.be.eq(0);
+        });
+      });
+
+      context('when the pool pays yield fees', () => {
+        it('is set on initialization', async () => {
+          await pool.init({ initialBalances });
+
+          const rates = pool.weights.map(() => fp(1));
+          const expectedRateProduct = calculateInvariant(rates, pool.weights);
+          expect(await pool.instance.getATHRateProduct()).to.be.almostEqual(expectedRateProduct, 0.0000001);
+        });
+
+        context('once initialized', () => {
+          sharedBeforeEach('initialize pool', async () => {
+            await pool.init({ initialBalances, recipient: lp });
+          });
+
+          context('when rates increase', () => {
+            let expectedRateProduct: BigNumber;
+            sharedBeforeEach('accumulate fees by increasing balance', async () => {
+              const rates = rateProviders.map(() => fp(random(1.0, 5.0)));
+
+              for (const [index, provider] of rateProviders.entries()) {
+                if (typeof provider !== 'string') await provider.mockRate(rates[index]);
+              }
+
+              expectedRateProduct = calculateInvariant(rates, pool.weights);
+            });
+
+            it('is updated by joins', async () => {
+              // We only test with a proportional join, since all joins are treated equally
+              await pool.join({
+                data: WeightedPoolEncoder.joinAllTokensInForExactBPTOut((await pool.totalSupply()).div(2)),
+                from: lp,
+              });
+
+              expectEqualWithError(await pool.instance.getATHRateProduct(), expectedRateProduct);
+            });
+
+            it('is updated by exits', async () => {
+              // We only test with a proportional exit, since all exits are treated equally and proportional exits remain
+              // enabled while paused
```

Comment:
Proportional exits are no longer enabled while paused - and why would that be relevant?